echoregions/tests/test_regions2d.py

```diff
@@ -454,41 +454,44 @@ def test_mask_correct_labels(
         DataArray containing Sv data of test zarr file.
     """
 
-    # Extract Region IDs and convert to Python float values
-    region_ids = regions2d_fixture.data.region_id.astype(float).to_list()
+    # Attempt to create mask on region with invalid depth values
+    M = regions2d_fixture.mask(da_Sv_fixture.isel(channel=0), [8])
 
-    # Create mask.
-    M = regions2d_fixture.mask(
-        da_Sv_fixture.isel(channel=0), region_ids, mask_labels=region_ids
-    )
+    # Check that all values are null
+    assert M is None
 
-    # Check that the mask's values matches only 13th and 18th region and there exists a nan value
-    # and that there exists a point of no overlap (nan value)
-    values = list(np.unique(M))
-    assert values[0] == 13.0
-    assert values[1] == 18.0
-    assert np.isnan(values[2])
+    # Create mask with regions that have no overlap with the Sv Data Array
+    M = regions2d_fixture.mask(da_Sv_fixture.isel(channel=0), [8, 9, 10])
+    M.isnull().all()
 
 
 @pytest.mark.regions2d
-def test_select_type_error(regions2d_fixture: Regions2D) -> None:
+def test_mask_correct_labels(
+    regions2d_fixture: Regions2D, da_Sv_fixture: DataArray
+) -> None:
     """
-    Test for select region errors.
+    Test if the generated region_id values are as expected
 
     Parameters
     ----------
     regions2d_fixture : Regions2D
         Object containing data of test EVR file.
+    da_Sv_fixture : DataArray
+        DataArray containing Sv data of test zarr file.
     """
 
-    # Check empty dataset error
-    with pytest.raises(TypeError):
-        empty_dataset = Dataset()
-        _ = regions2d_fixture.select_region(empty_dataset)
-    # Check empty tuple error
-    with pytest.raises(TypeError):
-        empty_tuple = ()
-        _ = regions2d_fixture.select_region(empty_tuple)
+    # Extract Region IDs and convert to Python float values
+    region_id = regions2d_fixture.data.region_id.astype(int).to_list()
+
+    # Create mask.
+    mask_3d = regions2d_fixture.mask(
+        da_Sv_fixture.isel(channel=0), region_id=region_id, mask_name="mask"
+    )
+
+    # Check that the mask's region_id values match the expected masked regions
+    mask_3d_region_id = mask_3d.region_id
+    assert mask_3d_region_id[0] == 13
+    assert mask_3d_region_id[1] == 18
```

Comment:
Why does the first region (`region_id=9`) disappear here?
It seems that the time and depth are within the acceptable range?

Reply:
The minimum time from the Sv ping time is:
```
<xarray.DataArray 'ping_time' ()>
array('2019-07-02T18:40:00.000000000', dtype='datetime64[ns]')
```

From `transect_multi_mask.evr`, we have the following:
```
13 4 9 0 4 -1 1 20190702 1243084826  9.2447583998 20190702 1314011892  758.9732173069
0
0
Off-transect
20190702 1243084826 10 20190702 1243084826 758.9732173069 20190702 1314011892 758.9732173069 20190702 1314011892 9.2447583998 0
Region 9
```
So we have times below the minimum ping time here.

Reply:
Oh I see, you're right. I failed to check the timestamps...